### PR TITLE
Added internationalization to rent created email notifications

### PR DESCRIPTION
**Pull request by tagonzalez**
## Summary
 [Internacionalizar el Email](https://trello.com/c/4VyymEzk/24-internacionalizar-el-email)

Added internationalization to the rent created user notification email.
## Notes
- Added `locale` column to User model using a migration
- Added `rails-i18n` gem to get spanish date formats

### Spanish emails
spanish-mails

### English emails
english-emails

### CODE REVIEW

File: app/mailers/user_mailer.rb [hunk 1]
```diff
@@ -4,14 +4,13 @@ class UserMailer < ApplicationMailer
...
     @user = @rent.user
-    @book = @rent.book
 
-    mail(
-      to: @user.email,
-      subject: 'A new rent has been created!'
-    ) do |format|
-      format.text
-      format.html
+    I18n.with_locale(@user.locale) do
```
Comment (mcolombo87) [inline app/mailers/user_mailer.rb]: You can define "I18n.locale" from the controller (with an "before_action") and don't do this... in this way, that definition can be reused in another place.

File: db/migrate/20190805181621_add_locale_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddLocaleToUsers < ActiveRecord::Migration[5.1]
+  def change
+    add_column :users, :locale, :string, null: false, default: :en
```
Comment (mcolombo87) [inline db/migrate/20190805181621_add_locale_to_users.rb]: It's ok define this field how "null: false" and set a default. So, if you are defining a "not null" maybe you must create a validator in the model ¿not?

Comment (tagonzalez) [reply db/migrate/20190805181621_add_locale_to_users.rb]: You're right!
Presence validator for `locale` was added in latest commits.

File: app/mailers/user_mailer.rb (hunk 1 above)
Comment (tagonzalez) [reply app/mailers/user_mailer.rb]: Great, it's cleaner that way. 
Changes made in latest commits.

File: app/controllers/api/v1/rents_controller.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@ module Api
   module V1
     class RentsController < ApplicationController
       before_action :authenticate_user!
+      before_action :set_locale
```
Comment (mcolombo87) [inline app/controllers/api/v1/rents_controller.rb]:
```suggestion
      before_action :authenticate_user!, :set_locale
```

File: app/controllers/api/v1/rents_controller.rb [hunk 4]
```diff
@@ -20,6 +21,10 @@ def create
         end
       end
 
+      def set_locale
+        I18n.locale = current_user.locale
```
Comment (mcolombo87) [inline app/controllers/api/v1/rents_controller.rb]:
```suggestion
        I18n.locale = current_user&.locale || I18n.default_locale
```
Look this!! http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

**Pull request merged**